AK/Tests/TestSpan.cpp

```diff
@@ -109,25 +109,25 @@ TEST_CASE(at_and_index_operator_return_same_value)
 
 TEST_CASE(can_subspan_whole_span)
 {
-    u8 buffer[16];
-    Bytes bytes { buffer, 16 };
+    static constexpr u8 buffer[16] {};
+    constexpr ReadonlyBytes bytes { buffer, 16 };
 
-    Bytes slice = bytes.slice(0, 16);
+    constexpr auto slice = bytes.slice(0, 16);
 
-    EXPECT_EQ(slice.data(), buffer);
-    EXPECT_EQ(slice.size(), 16u);
+    static_assert(slice.data() == buffer);
+    static_assert(slice.size() == 16u);
 }
 
 TEST_CASE(can_subspan_as_intended)
 {
-    const u16 buffer[8] { 1, 2, 3, 4, 5, 6, 7, 8 };
+    static constexpr u16 buffer[8] { 1, 2, 3, 4, 5, 6, 7, 8 };
 
-    Span<const u16> span { buffer, 8 };
-    auto slice = span.slice(3, 2);
+    constexpr Span<const u16> span { buffer, 8 };
+    constexpr auto slice = span.slice(3, 2);
 
-    EXPECT_EQ(slice.size(), 2u);
-    EXPECT_EQ(slice[0], 4);
-    EXPECT_EQ(slice[1], 5);
+    static_assert(slice.size() == 2u);
+    static_assert(slice[0] == 4);
+    static_assert(slice[1] == 5);
```

Comment:
Why are you replacing everywhere `EXPECT_EQ(lhs, rhs)` with `static_assert`?

This is a completely different behaviour: `EXPECT_EQ` just prints a message if `lhs != rhs`. `static_assert` will stop compilation if the condition is false.

Reply:
The `static_assert` executes at compile-time which is needed to test that the `constexpr` works. The `static_assert` will also print a message, just a slightly different one. 

```
/AK/Tests/TestSpan.cpp:52:5: error: static_assert failed due to requirement 'bytes[2] == 4'                                                                                               
    static_assert(bytes[2] == 4);                                                                        
    ^             ~~~~~~~~~~~~~                                                                          
1 error generated.                                                                                       
make[2]: *** [AK/Tests/CMakeFiles/TestSpan.dir/build.make:63: AK/Tests/CMakeFiles/TestSpan.dir/TestSpan.cpp.o] Error 1                                                                                            
```

Reply:
> The static_assert will also print a message, just a slightly different one.

Yes, but at compile time, not at runtime.

Reply:
Ah, we also use `static_assert` in `AK/Tests/TestArray.cpp`, so I guess it's fine. I didn't know.